pkg/operator/ocp/ensure.go

```diff
@@ -0,0 +1,58 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ocp
+
+import (
+	"context"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/subctl/pkg/clusterrole"
+	"github.com/submariner-io/subctl/pkg/clusterrolebinding"
+	"github.com/submariner-io/submariner-operator/pkg/embeddedyamls"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+func EnsureRBAC(dynClient dynamic.Interface, kubeClient kubernetes.Interface, namespace string) (bool, error) {
+	if !IsOcpPlatform(dynClient) {
+		return false, nil
+	}
+
+	createdCR, err := clusterrole.EnsureFromYAML(kubeClient, embeddedyamls.Config_rbac_submariner_operator_cluster_role_yaml)
+	if err != nil {
+		return false, errors.Wrap(err, "error provisioning the operator OCP ClusterRole resource")
+	}
+
+	createdCRB, err := clusterrolebinding.EnsureFromYAML(kubeClient, namespace,
+		embeddedyamls.Config_rbac_submariner_operator_ocp_cluster_role_binding_yaml)
+	if err != nil {
+		return false, errors.Wrap(err, "error provisioning the operator OCP ClusterRoleBinding resource")
+	}
+
+	return createdCR || createdCRB, errors.Wrap(err, "error updating the SCC resources")
```

Comment:
Here the booleans are ored, whereas in `submariner/ocp` they are anded — which should it be?

Reply:
It should be and

Reply:
Hmm, but even if one of clusterrole , clusterrolebinding was created, we won't have privileged permission for the component, what am I missing here?

Reply:
The purpose of the return boolean is to report if something was actually created (successfully), ie if it didn't already exist, so the caller can report that to the user. A resource may already exist if `join` was previously run. The return error indicates if a creation failed.

Reply:
Ack, 10x. will change to or